astropy/units/decorators.py

```diff
@@ -115,16 +116,38 @@ def wrapper(*func_args, **func_kwargs):
                 # If the target unit is empty, then no unit was specified so we
                 # move past it
                 if target_unit is not funcsigs.Parameter.empty:
+
+                    if isinstance(target_unit, str):
+                        from .core import Unit
```

Comment:
Oops, yes. I ran into a circular import issue and was debugging, but forgot to remove this.